test/test.lua

```diff
@@ -1634,6 +1634,93 @@ function test.multi_gpu_random()
    cutorch.setDevice(1) -- reset device
 end
 
+function test.multinomial_with_replacement()
+   for tries = 1, 10 do
+      local n_row = torch.random(10)
+      local n_col = 1 + torch.random(1000)
+
+      local prob_dist = torch.CudaTensor(n_row, n_col):uniform()
+      prob_dist:select(2, n_col):fill(0) --index n_col shouldn't be sampled
+      local n_sample = torch.random(n_col - 1)
+      local sample_indices = torch.multinomial(prob_dist, n_sample, true)
+      tester:assert(prob_dist:dim() == 2, "wrong number of prob_dist dimensions")
```

Comment:
Why are you checking for prob_dist? You've created that yourself.
Wouldn't it be more useful to check that sample_indices has the right number of dimensions and the right size?